SetupDataPkg/Tools/VariableList.py

```diff
@@ -1278,6 +1277,68 @@ def write_csv(schema, csv_path, full, subknobs=True):
                             knob.format.object_to_string(knob.value),
                             string_binary,
                             knob.help])
+                            
+                            
+def write_csv_detailed(schema, csv_path):
+
+    with open(csv_path, 'w', newline='') as csv_file:
+        writer = csv.writer(csv_file)
+        guid = None
+        namespace_changed = True # Used to print the guid on the first row and then only print
+                                 # another guid if it changes
+        
+        writer.writerow(['Guid', 'Knob', 'Current Value', 'Binary', 'Default Value', 'Valid Value Options', 'Help'])

```

Comment:
Have you tested reading this format back in? Are the new fields ignored and so the existing reading function works? If I understand correctly, you are also splitting the knobs different than the existing writing function, does the reading function understand this?

Reply:
I have not modified any of the pre-existing functions, so all should be intact and working fine with all other scripts/functions that consume CSV files.
Testing if the CSV file generated from the new function I developed can be read correctly by other scripts/functions in the repo is WIP. In the worst case, I will need to add new functions to consume this new format. Existing functions should not be impacted.